test/Chainweb/Test/Pact/RemotePactTest.hs

```diff
@@ -227,6 +230,38 @@ localContTest iot nio = testCaseSteps "local continuation test" $ \step -> do
         $ mkCont
         $ mkContMsg pid 1
 
+pollingConfirmDepth :: IO (Time Micros) -> IO ChainwebNetwork -> TestTree
+pollingConfirmDepth iot nio = testCaseSteps "poll confirmation depth test" $ \step -> do
+    cenv <- _getServiceClientEnv <$> nio
+
+    step "execute /send with initial pact continuation tx"
+    cmd1 <- firstStep
+    rks <- sending sid cenv (SubmitBatch $ pure cmd1)
+
+    step "check /poll responses to extract pact id for continuation"
+
+    beforePolling <- getCurrentBlockHeight v cenv sid
+    PollResponses _ <- pollingWithDepth sid cenv rks (Just $ ConfirmationDepth 70) ExpectPactResult
+    afterPolling <- getCurrentBlockHeight v cenv sid
+
+    case (beforePolling, afterPolling) of
+      (Just bf, Just af) ->
+        assertBool "the difference between heights should be no less than the confirmation depth" $ (af - bf) >= 70
+      _ -> assertFailure "couldn't get block heights"
```

Comment:
It took some looking but I see what you're doing here. You rely on the `pollingWithDepth` function doing a bunch of retries until the poll succeeds. So this test checks the current block height, polls until the tx goes in, and then checks again; if the confirmation depth param worked, we should've seen at least `depth` blocks pass while polling. Maybe this test could use a comment explaining the whole idea, but I think the `step` suggestions above are enough.

It does take a while, like 14 seconds. But I really like the test, so that's fine with me.

Reply:
Yeah, I have reduced the confirmation depth to `10`, takes a few seconds instead of `14` now.